lib/teiserver/account/tasks/delete_user_task.ex

```diff
@@ -55,16 +60,17 @@ defmodule Teiserver.Admin.DeleteUserTask do
       "DELETE FROM teiserver_game_rating_logs WHERE user_id = ANY($1)",
 
       # Moderation
-      "DELETE FROM moderation_reports WHERE reporter_id = ANY($1) OR target_id = ANY($1) OR responder_id = ANY($1)",
-      "DELETE FROM moderation_actions WHERE target_id = ANY($1)"
+      "DELETE FROM moderation_reports WHERE reporter_id = ANY($1)",
+      "DELETE FROM moderation_reports WHERE target_id = ANY($1)",
+      "DELETE FROM moderation_actions WHERE responder_id = ANY($1)"
     ]
     |> Enum.each(fn query ->
-      Ecto.Adapters.SQL.query!(Repo, query, [id_list])
+      Ecto.Adapters.SQL.query(Repo, query, [int_id_list])
     end)
 
     # And now the users
     query = "DELETE FROM account_users WHERE id = ANY($1)"
-    Ecto.Adapters.SQL.query!(Repo, query, [id_list])
+    Ecto.Adapters.SQL.query!(Repo, query, [int_id_list])
```

Comment:
>I have problems with the delete query raising errors as some tables didn't exist in my setup (e.g. telemetry). I therefore switched to Ecto.Adapters.SQL.query instead of Ecto.Adapters.SQL.query! to not raise errors.

You should ensure your setup is correct first. There are some guide in the repo, and you might be interested in [postgres setup](https://github.com/beyond-all-reason/teiserver/blob/master/documents/guides/local_setup.md)

I have the telemetry tables locally and I didn't recall doing anything fancy to get them.

The main problem with swapping `query!/3` for `query/3` is that it doesn't raise an error and returns the exception as a value, but that doesn't fix the underlying problem.

If anything, one thing to fix here would be to wrap all the deletes inside a transaction, but that's irrelevant for your changes.

Reply:
I did recreate my setup with the guide once again to make sure I did not miss anything.
However, the some columns still do not exist.
I observed that all ```telemetry_\*_event_types``` only have ```["id", "name"]```.
That makes sense, considering they only define the event types, not when events happen to users.
The ```telemetry_\*_events``` contain the ```user_id```, therefore I think this was originally meant. 

I also changed some queries, because the columns are named differently.

I don't have any problem using ```query!``` now

<details>
<summary>To look at the columns I used something like this:</summary>
<br>

```
[
"SELECT * FROM moderation_actions"
]
|> Enum.each(fn query ->
Ecto.Adapters.SQL.query!(Repo, query) |> IO.inspect()
end)
```
</details>